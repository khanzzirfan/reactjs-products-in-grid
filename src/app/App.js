import 'bootstrap/dist/css/bootstrap.css'
import React from 'react';
import s from './App.scss';
import { Link } from 'react-router-dom';
import Routes from '../modules/routes';
import AppNavbarComponent from 'src/common/app-navbar-component';
import AppHeaderComponent from 'src/common/app-header-component';
import AppContainer from 'src/products/ui/containers/app-container';

const App = () =>
    <div>
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
            <AppContainer />
        </nav>
        <div className="container main-content">
            {Routes}
        </div>
    </div>;


export default App;
