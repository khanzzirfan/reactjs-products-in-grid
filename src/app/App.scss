@import '../variables.scss';
 
/*
 * Base styles
 * ========================================================================== */
 
html {
  color: #222;
  font-weight: 100;
  font-size: 1em; /* ~16px; */
  font-family: $font-family-base;
  line-height: 1.375; /* ~22px */
}
 
.container {
  margin: 0 auto;
  padding: 0 0 40px;
  max-width: $max-content-width;
}


/* Small devices (tablets, 768px and up) */
@media screen and (min-width: 200px) {
  .main-content{
    padding-top: 250px;
  }
}

@media screen and (min-width: 500px) {
  .main-content{
    padding-top: 95px;
  }
}