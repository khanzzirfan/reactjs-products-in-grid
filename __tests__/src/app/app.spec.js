import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from 'src/app/app.js';
import AppContainer from 'src/products/ui/containers/app-container.js';

describe("app test", () => {

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<App />);
    });

    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("it should render with correct class names", ()=>{
       
        expect(wrapper.find('.fixed-top').exists()).toBeTruthy();
        expect(wrapper.find('.navbar-dark').exists()).toBeTruthy();
        expect(wrapper.find('.navbar-expand-lg').exists()).toBeTruthy();
       
    });

    it(" should render component with app container component", () => {
        expect(wrapper.find(AppContainer).exists()).toBeTruthy();
    });
});