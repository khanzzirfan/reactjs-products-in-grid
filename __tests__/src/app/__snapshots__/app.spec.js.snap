// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test component should match to snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  >
    <Connect(AppContainer) />
  </nav>
  <div
    className="container main-content"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
    </Switch>
  </div>
</div>
`;
